akka-http-core/src/main/scala/akka/http/impl/engine/server/ServerTerminator.scala

```diff
@@ -92,7 +111,13 @@ private[http] final class MasterServerTerminator(log: LoggingAdapter) extends Se
         if (terminators.compareAndSet(v, Terminating(timeout.fromNow))) {
           // cause the termination for all connections
           val connectionsTerminated = Future.sequence(ts.map { t ⇒
-            t.terminate(timeout)
+            // termination in general always succeeds, but we make sure here in order
+            // to not accidentally short-circuit terminating the other connection-terminators -- all must be terminated
+            t.terminate(timeout).recover {
+              case ex ⇒
+                log.warning("Ignoring termination failure of {}, failure was: {}", t, ex.getMessage)
```

Comment:
what will `t.toString` be here? a Future with some collection? All connections? Will that be a good log message?